### PR TITLE
Feature: Built Branch & Release Action

### PR DESCRIPTION
**Pull request by srtfisher**
Moves the [previous reusable workflow](https://github.com/alleyinteractive/.github?tab=readme-ov-file#built-releases) to a standalone action. Supports any future plans for improving the built releases including better change logs.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    required: false
+    default: 'npm run build'
+  draft:
+    description: 'Specify whether to create a draft release.'
+    required: false
+    default: 'false'
+  github-token:
+    description: 'Specify the GitHub token to use.'
+    required: false
+    default: ''
```
Comment (mogmarsh) [inline action.yml]: should this match the readme?
```suggestion
    default: ${{ github.token }}
```

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+      with:
+        php-version: '${{ inputs.php-version }}'
+        skip-audit: 'true'
+        skip-services: 'true'
+        skip-test: 'true'
+        skip-wordpress-install: 'true'
+
+    - name: NPM Install and Build
+      uses: alleyinteractive/action-test-node@develop
+      if: steps.extract-has-node-assets.outputs.value == 'true'
```
Comment (mogmarsh) [inline action.yml]: do we need a `skip-npm-install` check?

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+      if: steps.extract-is-built-branch.outputs.value == 'false' && steps.extract-branch.outputs.value != 'false' && github.repository != 'alleyinteractive/create-wordpress-plugin'
+      shell: bash
+      env:
+        CURRENT_BRANCH: '${{ steps.extract-branch.outputs.value }}'
+        CREATING_RELEASE: ${{ steps.extract-version.outputs.value != 'false' && steps.extract-tag-exists.outputs.value == 'false' && 'true' }}
+      run: |
+        git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+        git config --global user.name "$GITHUB_ACTOR"
+
+        rm -rf .gitignore docker_tag output.log
```
Comment (mogmarsh) [inline action.yml]: should we leave .gitignore if there's no `.deployignore` and no `.distignore`? 🤔

**Review by mogmarsh — APPROVED** [review]
a couple questions, otherwise 👍

File: action.yml (hunk 1 above)
Comment (srtfisher) [reply action.yml]: I was doing the check down below but sure

File: action.yml (hunk 3 above)
Comment (srtfisher) [reply action.yml]: true that

**Pull request merged**